[Media] Fix RemoteMediaPlayerProxy::stopVideoFrameMetadataGathering typo

youssefsoli opened the pull request:
#### 7a5f68dbe338a1e637ec892668335e8885d63ae3
<pre>
[Media] Fix RemoteMediaPlayerProxy::stopVideoFrameMetadataGathering typo
<a href="https://bugs.webkit.org/show_bug.cgi?id=243420">https://bugs.webkit.org/show_bug.cgi?id=243420</a>
&lt;rdar://97932373&gt;

Reviewed by Eric Carlson and Jer Noble.

There is a typo in RemoteMediaPlayerProxy::stopVideoFrameMetadataGathering
that calls m_player-&gt;startVideoFrameMetadataGathering rather than
m_player-&gt;stopVideoFrameMetadataGathering, potentially causing premature
metadata gathering.

* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.cpp:
(WebKit::RemoteMediaPlayerProxy::stopVideoFrameMetadataGathering):

Fixed up the assertions to catch startVideoFrameMetadataGathering from
being called twice.
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::startVideoFrameMetadataGathering):
* Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm:
(WebCore::MediaPlayerPrivateWebM::startVideoFrameMetadataGathering):

Canonical link: <a href="https://commits.webkit.org/253011@main">https://commits.webkit.org/253011@main</a>
</pre>

eric-carlson reviewed (approved):
Great catch!

r=me

youssefsoli committed:
[Media] Fix RemoteMediaPlayerProxy::stopVideoFrameMetadataGathering typo
https://bugs.webkit.org/show_bug.cgi?id=243420
<rdar://97932373>

Reviewed by Eric Carlson and Jer Noble.

There is a typo in RemoteMediaPlayerProxy::stopVideoFrameMetadataGathering
that calls m_player->startVideoFrameMetadataGathering rather than
m_player->stopVideoFrameMetadataGathering, potentially causing premature
metadata gathering.

* Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.cpp:
(WebKit::RemoteMediaPlayerProxy::stopVideoFrameMetadataGathering):

Fixed up the assertions to catch startVideoFrameMetadataGathering from
being called twice.
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::startVideoFrameMetadataGathering):
* Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm:
(WebCore::MediaPlayerPrivateWebM::startVideoFrameMetadataGathering):

Canonical link: https://commits.webkit.org/253011@main
